src/Processor/Braintree/Mapper/MapsAddresses.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace TeamGantt\Dues\Processor\Braintree\Mapper;
+
+use Braintree\Address as BraintreeAddress;
+use Braintree\Transaction\AddressDetails;
+use TeamGantt\Dues\Model\Address;
+use TeamGantt\Dues\Model\Address\Country;
+use TeamGantt\Dues\Model\Address\State;
+use Throwable;
+
+trait MapsAddresses
+{
+    /**
+     * @param BraintreeAddress|AddressDetails $braintreeAddress
+     */
+    protected function toAddress($braintreeAddress): Address
```

Comment:
8.1 gives us union types now. Not sure if we wanna start doing that now, or wait a bit?

Reply:
I would be game to wait for a bit. The dues container is still rocking a 7.4 version.

A php 8.1 task would be great for the future though. We can bring the container up to date, add union types, and deprecate the spatie/enum package in favor of native enums. Will be great!